openerp/addons/base/res/res_users.py

```diff
@@ -548,8 +549,19 @@ def preference_change_password(self, cr, uid, ids, context=None):
             'target': 'new',
         }
 
-    @tools.ormcache(skiparg=2)
+    @api.v7
     def has_group(self, cr, uid, group_ext_id):
+        return self._has_group(cr, uid, group_ext_id)
+    @api.v8
+    def has_group(self, group_ext_id):
+        # use singleton's id if called on a non-empty recordset, otherwise
+        # context uid
+        uid = self.id if self else self.env.uid
```

Comment:
With my nit-picker hat on, I find `uid = self.id or self.env.uid` slightly more readable. And it does not use the supermagical `if self` that looks plain crazy in this context. It should behave in the same manner, shouldn't it? (i.e. fails if `self` ever holds several records)